extensions/gdx-tools/src/com/badlogic/gdx/tools/particleeditor/ParticleEditor.java

```diff
@@ -428,6 +428,11 @@ public void render () {
 			Gdx.gl.glClearColor(colors[0], colors[1], colors[2], 1.0f);
 			Gdx.gl.glClear(GL20.GL_COLOR_BUFFER_BIT);
 
+			//TODO: lwjgl3 currently calls the render method too early
+			if(previewImagePanel==null){

```

Comment:
Formatting!

Reply:
I hope I can get rid of this again, I think there is something wrong in the initialization order currently

Reply:
I found a fix for this: https://github.com/tisoft/libgdx/pull/1.

Reply:
Thanks!

I'll try to make the Lwjgl3AWTCanvas a drop in replacement for the Lwjgl2 stuff, so I hope I can get around the issue, without changing the ParticleEditor. But if I don't manage to do that, I'll just take your PR. 😄

Reply:
Sounds good 👍

Reply:
I used your reorder fix now. :)

In Lwjgl2, there are two differnt implementations: `LwjglCanvas` and `LwjglAWTCanvas`. I used the `LwjglAWTCanvas` as base for the new `Lwjgl3AWTCanvas`. The `ParticleEditor` used the `LwjglCanvas`, though. Replacing that with the `LwjglAWTCanvas` gives the same NPE as with my new `Lwjgl3AWTCanvas`. So in deed the `Lwjgl3AWTCanvas` is a drop in replacement for `LwjglAWTCanvas`, but probably not for `LwjglCanvas`.